clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -290,17 +315,47 @@ public void poll(long timeout) throws IOException {
 
         clear();
 
-        if (hasStagedReceives() || !immediatelyConnectedKeys.isEmpty())
+        if (hasStagedReceives() || !immediatelyConnectedKeys.isEmpty() || !keysWithBytesBuffered.isEmpty())
             timeout = 0;
 
+        boolean poolOutOfMemory = memoryPool.isOutOfMemory();
+        if (poolOutOfMemory && !outOfMemory) {
+            //dont bother selecting for reads
+            outOfMemory = true;
+            log.trace("Broker is low on memory - muting incoming sockets");
```

Comment:
personally I dont think this is a warning - its normal operations. users who care can get at this information in a much better way via the sensors exposed in this KIP.